add "save_when" to ios_banner

lvrfrc87 opened the pull request:
##### SUMMARY

I have add "save_when" functionality, taken from ios_config. However ios_banner arguments need to be updated. I have tried to reverse engineering the code to understand where the argument list has to be updated but I could not find the bottom of it
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ios_banner

lvrfrc87 committed:
add "save_when"

I have add "save_when" functionality, taken from ios_config. However ios_banner arguments need to be updated. I have tried to reverse engineering the code to understand where the argument list has to be updated but I could not find the bottom of it

FragmentedPacket commented:
https://github.com/ansible/ansible/blob/99946cdab9cf4b7a5676ba2e64001c51f65eaae3/lib/ansible/modules/network/ios/ios_banner.py#L175-L186

I'd make a few changes:

- I would move the lines above (L175-186) to below L204 with the correct indentation.
- Remove `ignore_lines=diff_ignore_lines` from L180/181
- Remove the extra `)` from L164


Also, what errors are you receiving when running this modified module?

FedericoOlivieri commented:
The error I get is:

`FAILED! => {"changed": false, "msg": "Unsupported parameters for (ios_banner) module: save_when Supported parameters include: auth_pass, authorize, banner, host, password, port, provider, ssh_keyfile, state, text, timeout, username"}`

lvrfrc87 committed:
Update ios_banner 

Move the lines above (L175-186) to below L204 with the correct indentation.
Remove ignore_lines=diff_ignore_lines from L180/181
Remove the extra ) from L164

lvrfrc87 committed:
fix line 163

lvrfrc87 committed:
Update default and fix indentation

lvrfrc87 committed:
fix space

ib/ansible/modules/network/ios/ios_banner.py:107:1: E302 expected 2 blank lines, found 0
lib/ansible/modules/network/ios/ios_banner.py:121:1: E302 expected 2 blank lines, found 0
lib/ansible/modules/network/ios/ios_banner.py:138:1: E302 expected 2 blank lines, found 0
lib/ansible/modules/network/ios/ios_banner.py:147:1: E302 expected 2 blank lines, found 0
lib/ansible/modules/network/ios/ios_banner.py:155:1: E302 expected 2 blank lines, found 0
lib/ansible/modules/network/ios/ios_banner.py:196:1: E305 expected 2 blank lines after class or function definition, found 0

lvrfrc87 committed:
fix pypi and pep8

abenokraitis commented:
Thanks for your submission, this probably isn't useful and the PR has been deleted. Please re-open and/or resubmit a new PR for review.
